pytorch_lightning/utilities/parsing.py

```diff
@@ -159,18 +166,21 @@ def collect_init_args(frame, path_args: list, inside: bool = False) -> list:
           most specific class in the hierarchy.
     """
     _, _, _, local_vars = inspect.getargvalues(frame)
-    if '__class__' in local_vars:
-        local_args = get_init_args(frame)
-        # recursive update
-        path_args.append(local_args)
-        return collect_init_args(frame.f_back, path_args, inside=True)
-    elif not inside:
-        return collect_init_args(frame.f_back, path_args, inside)
+    if isinstance(frame.f_back, FrameType):
```

Comment:
mind comment on why this?

Reply:
Dropping this condition leads to the following two issues:
```
utilities/parsing.py:174: error: Argument 1 to "collect_init_args" has incompatible type "Optional[FrameType]"; expected "FrameType"
utilities/parsing.py:176: error: Argument 1 to "collect_init_args" has incompatible type "Optional[FrameType]"; expected "FrameType"
```

Maybe, there's another nice workaround.